lib/aliases.js

```diff
@@ -48,49 +46,45 @@ module.exports = function(polyfillNameResolverFunctions) {
  * @return {array} An expanded list of polyfill identifiers
  */
 function applyResolverToIdentifiers(featureSet, nameResolverFunction) {
-	let newNamesDiscovered = false;
+	const processQueue = Object.keys(featureSet);
 
-	function run() {
-		return Promise.all(Object.keys(featureSet).map(featureName => {
-			const feature = featureSet[featureName];
-			return Promise.resolve(nameResolverFunction(featureName)).then(resultItems => {
+	while (processQueue.length > 0) {
+		const featureName = processQueue.shift();
+		const feature = featureSet[featureName];
+		const resultItems = nameResolverFunction(featureName);
 
-				// No aliases for this feature - do not modify feature set
-				if (!Array.isArray(resultItems)) return true;
-				if (resultItems.length === 1 && resultItems[0] === featureName) return true;
+		// No aliases for this feature - do not modify feature set
+		if (!Array.isArray(resultItems)) continue;
+		if (resultItems.length === 1 && resultItems[0] === featureName) continue;
 
-				resultItems.forEach(function(newName) {
-					if (newName !== featureName) {
-						const result = Object.assign({}, feature, {
-							aliasOf: uniq([featureName].concat(feature.aliasOf || [])),
-							flags: feature.flags || []
-						});
+		resultItems.forEach(function(newName) {
+			if (newName !== featureName) {
 
-						if (!featureSet[newName]) {
-							featureSet[newName] = result;
-							newNamesDiscovered = true;
-						} else {
+				const props = {
+					aliasOf: new Set([featureName, ...(feature.aliasOf || [])]),
+					flags: new Set([...(feature.flags || [])])
+				};
 
-							// Merge properties
-							['flags', 'aliasOf'].forEach(prop => {
-								featureSet[newName][prop] = (prop in featureSet[newName]) ? uniq(featureSet[newName][prop].concat(result[prop])).sort() : result[prop];
-							});
-						}
-					}
-				});
+				if (!featureSet[newName]) {
+					featureSet[newName] = Object.assign({}, feature, props);
+					processQueue.push(newName);
+				} else {
 
-				// If alias resolver returned a list that did not include the original name, remove that from the feature set
-				if (resultItems.indexOf(featureName) === -1) delete featureSet[featureName];
-			});
-		})).then(function() {
-			if (newNamesDiscovered) {
-				newNamesDiscovered = false;
-				return run();
```

Comment:
Speedup: previously we would re-iterate the entire collection if any new items were added.  Now we add new items to the end of a queue, and only process the new ones.